packages/manager/apps/pci-volume-snapshot/src/api/hooks/useSnapshots.ts

```diff
@@ -0,0 +1,68 @@
+import { useMemo } from 'react';
+import { useQuery, useQueries } from '@tanstack/react-query';
+import { ColumnSort, PaginationState } from '@ovh-ux/manager-react-components';
+import { applyFilters, Filter } from '@ovh-ux/manager-core-api';
+import { getSnapshots, getVolume, TSnapshot, TVolume } from '../data/snapshots';
+import { paginateResults, sortResults } from '@/helpers';
+
+export type TVolumeSnapshot = TSnapshot & {
+  volume?: TVolume | null;
+};
+
+export const useAllSnapshots = (projectId: string) =>
+  useQuery({
+    queryKey: ['snapshots', projectId],
+    enabled: !!projectId,
```

Comment:
If you make this predicate, I guess `projectId` parameter's type in missing `| undefined` :)

Reply:
It can't be undefined but it can be empty (which is falsy too)